examples/vision/video_classification.py

```diff
@@ -0,0 +1,466 @@
+layers must have a time-step dimension. So, how do we incorporate that in the extracted
+features?
+
+The features have a shape of `(len(train_frames), 5, 5, 2048)`. Recall that each video is
+now represented as an ordered sequence of **five** frames. So, we can reshape our features
+like so: `(len(train_frames) / 5, 5, 5 * 5 * 2048)`. The second dimension (5) is now the
+time-step dimension. It turns out that this heuristic works quite well in practice. It
+just needs as adjustment in the number of examples such that it is divisible by the
+number of frames representing a video. Fortunately, for this example, we did not have to
+do that explicitly.
+"""
+
+train_features = train_features.reshape(
+    len(train_features) // SEQ_LENGTH, SEQ_LENGTH, 5 * 5 * 2048
+)
+test_features = test_features.reshape(
+    len(test_features) // SEQ_LENGTH, SEQ_LENGTH, 5 * 5 * 2048
+)
+
+"""
+We now need to prepare the labels (the action classes), which are encoded as strings.
+Neural networks do not understand string values, so they must be converted to some numerical
+form before they are fed to the model. Here we will use the
+[`StringLookup`](https://keras.io/api/layers/preprocessing_layers/categorical/string_lookup)
+layer to binarize the string labels. It will first map the strings to integer indices and then it will
+[one-hot encode](https://developers.google.com/machine-learning/glossary#one-hot-encoding)
+those integer values.
+"""
+
+label_processor = keras.layers.experimental.preprocessing.StringLookup(
+    max_tokens=6, num_oov_indices=0, output_mode="binary", sparse=False
+)
+label_processor.adapt(new_train_df["class"].values)
+
+train_labels = label_processor(new_train_df["class"].values[..., None])
+test_labels = label_processor(new_test_df["class"].values[..., None])
+
+
+# We can verify the vocabulary like so.
+class_vocab = label_processor.get_vocabulary()
+print(class_vocab)
+
+# We can also verify if the string values were mapped as expected.
+idx = np.random.choice(len(train_labels))
+assert (
+    class_vocab[train_labels[idx].numpy().argmax(-1)]
+    == new_train_df["class"].values[idx]
+)
+
+"""
+We now need to skip some labels from both `train_labels` and `test_labels` since the
+number of samples has changed. 
+"""
+
+
+def prepare_labels(tags):
+    labels = []
+    for (i, label) in enumerate(tags):
+        if i % SEQ_LENGTH == 0:
+            labels.append(label)
+    return np.stack(labels)
+
+
+train_labels = prepare_labels(train_labels)
+test_labels = prepare_labels(test_labels)
+
+"""
+## Train the sequence model
+
+Empirically, [label-smoothing](https://arxiv.org/abs/1906.02629) was found to improve performance.
+This is why we represented the integer labels as one-hot encoded vectors in the first place.
+"""
+
+
+def get_sequence_model(optimizer, label_smoothing=0.1):
+    rnn_model = keras.Sequential(
+        [
+            keras.layers.GRU(
+                64, input_shape=(SEQ_LENGTH, 5 * 5 * 2048), return_sequences=True
```

Comment:
If you really want to provide an input shape, rather than using the `input_shape` argument prefer using an `Input()` object as first layer.